h/static/scripts/annotator/plugin/pdf-metadata.js

```diff
@@ -0,0 +1,73 @@
+'use strict';
+
+/**
+ * This PDFMetadata service provides implementations for extracting fields
+ * of a pdf document from the input PDFViewerApplication object.
+ *
+ * The fields extracted are:
+ * #1. urn - a string which uniquely identifies a pdf document.
+ * #2. title - a string representing the title of the pdf document.
+ * #3. link - an array of two objects, the urn and the url strings of the pdf.
+ * #4. documentFingerprint - the string representing the pdf fingerprint.
+ */
+function PDFMetadata(PDFViewerApplication) {
+  this._PDFViewerApplication = PDFViewerApplication;
+  this._documentLoaded = new Promise(function (resolve) {
+    var finish;
+    finish = function (evt) {
+      window.removeEventListener('documentload', finish);
+      return resolve();
+    };
+    return window.addEventListener('documentload', finish);
+  });
+};
+
+/**
+ * Returns a promise of a PDF URN string after the pdf loads
+ * succesfully and the 'documentload' event fires.
+ */
+PDFMetadata.prototype.getUrn = function () {
+  return this._documentLoaded.then(function () {
+    return 'urn:x-pdf:' + this._PDFViewerApplication.documentFingerprint;
+  }.bind(this));
+};
+
+/**
+ * Returns a promise of an object with the fields
+ * title(string), link(array) and documentFingerprint(string)
+ * after the pdf loads succesfully and the 'documentload' event fires.
+ */
+PDFMetadata.prototype.getMetadata = function () {
+  return this._documentLoaded.then(function () {
+    var info;
+    var metadata;
+    var urn;
+    var title;
+    var documentFingerprint;
+    var link = [];
+
+    info = this._PDFViewerApplication.documentInfo;
+    metadata = this._PDFViewerApplication.metadata;
+
+    if ((typeof metadata !== 'undefined' && metadata !== null ? metadata.has('dc:title') : void 0) &&
```

Comment:
I know this was ported from the original code but the convention we're using for more recently written code is just to use truthiness where possible to test for an object being null, so `if (metadata && metadata.has('dc:title')`. The rationale is this is more succinct and easier to read.